Spec/Tests/RealtimeClientChannelTests.swift

```diff
@@ -161,7 +173,67 @@ class RealtimeClientChannelTests: XCTestCase {
         static var failed = 0
         fileprivate init() {}
     }
+    
+    func test_deserialization_of_integers_with_JSONSerialization() {
+        
+        let jsonString = "{\"integerExample\": 1756699}"
+        let jsonStringData = Data(jsonString.utf8)
+        
+        let jsonObject = try! JSONSerialization.jsonObject(with: jsonStringData, options: .fragmentsAllowed) as! [String: Any]
+        
+        let doubleExample = jsonObject["integerExample"] as! Double
+        
+        print("Decoded Data: \(jsonObject), type: \(type(of: jsonObject)), as decimal: \(Decimal(doubleExample)), as double: \(doubleExample)")
+    }
+    
+    func test_deserialization_of_integers_with_JSONDecoder() {
+        
+        let jsonString = "{\"integerExample\": 1756699}"
+        let jsonStringData = Data(jsonString.utf8)
+        
+        let decodedObject = try! JSONDecoder().decode(Example.self, from: jsonStringData)
+        
+        print("Decoded Object: \(decodedObject), type: \(type(of: decodedObject)), as decimal: \(decodedObject.integerExample), as double: \(decodedObject.integerExample.doubleValue)")
```

Comment:
I ran it again, and to my amusement I've found out absence of fraction as well. I've already mentally prepared "to explain with gestures to a doctor why I have this light bulb in my mouth", but after I ran it a couple of times more I've finally got this results:

<img width="695" alt="Screenshot 2022-05-10 at 22 32 41" src="https://user-images.githubusercontent.com/45491869/167861110-19a982af-89d9-4df4-9066-6741ece28b81.png">

Xcode debugger is a weirdo... @lawrence-forooghian @lukasz-szyszkowski  @QuintinWillison

Reply:
That is very odd. I just ran the test 1000 times (iPhone 13 Pro simulator, iOS 15.4, Xcode 13.3.1) and did not get a single one with a fractional component. I wonder how I can reproduce this?

Reply:
> I've already mentally prepared "to explain with gestures to a doctor why I have this light bulb in my mouth"

I am deeply confused but I enjoyed it

Reply:
Ooh, now I ran it on an iPhone 12 (iOS 12.4) simulator and I get the decimal too! Weird.

Reply:
So, yeah, I guess there's some iOS version difference in how `JSONDecoder` handles `Decimal` values. But it seems to handle `Int` properly in both iOS 12 and 13.

Reply:
> > I've already mentally prepared "to explain with gestures to a doctor why I have this light bulb in my mouth"
> 
> I am deeply confused but I enjoyed it

The thing with light bulbs is that you can't evacuate it from your mouth by yourself (if you dared to check what will happen if you put it in you mouth in the first place 😆) Standard situation in EMT.

Reply:
I think it's the only way, because to incorporate `JSONDecoder` would require to add some swift sources to the Objective-C library and functionality will be accessible in swift only, so kinda not worth it.

Reply:
@QuintinWillison WDYT

Reply:
We have the exact same issue in our Java/Android SDK, where [GSON](https://github.com/google/gson) has been baked in as part of our public API. 😬 
(I raised an issue for it _some_ time ago 😳 ... yet to be worked on: https://github.com/ably/ably-java/issues/573)

I would much prefer that we simply expose JSON to users as an `NSData` instance, aligning with Apple's foundational JSON APIs. And, yes, that will need a spec update.

Reply:
https://github.com/ably/docs/issues/1425

Reply:
@maratal I think that if we want to move ably/docs#1425 forwards, we'll need to write the features spec change ourself; I don't think anybody else is going to do it. Are you happy to do that?

Reply:
> Are you happy to do that?
Sure, I'll do that.